launcher/tests/test_util.py

```diff
@@ -302,3 +302,56 @@ def test_get_logger():
     assert logger.name == test_prefix
     logger = util.get_logger(prefix=test_prefix, module=test_module)
     assert logger.name == "{}.{}".format(test_prefix, test_module)
+
+
+def test_is_sdapp_halted_runs_once():
+    """
+    When check for sd-app state is run
+     Then exactly one subprocess call should be run
+    """
+    with mock.patch("subprocess.check_output") as patched_subprocess_check:
+        util.is_sdapp_halted()
+        patched_subprocess_check.assert_called_once()
+
+
```

Comment:
This sub-process call doesn't have any side-effects, it is used _for the value it returns_.

From that point of view, I don't think it is needed or useful to assert that it is called, as long as the behavior of the function (`util.is_sdapp_halted`) is tested for all the possible return values of the sub-process call. And that's what you did with the two tests below. :+1: 

On the other size, these call assertions are a bit brittle: they tend to break if the number of calls or the parameters change, in ways that often don't reflect an actual change of behavior of the function under test ("false positives/negatives"), and get in the way of refactoring.

Because of that I think it's safe, and I'd suggest removing this test.